inst/htmlwidgets/plotly.js

```diff
@@ -2,29 +2,40 @@
 HTMLWidgets.widget({
   name: "plotly",
   type: "output",
-  
+
   initialize: function(el, width, height) {
-    return {};
+    return {
+      // Push JavaScript closures onto this list, and renderValue
+      // will pop them off and run them one at a time the next
+      // time it runs. Use this to dispose of e.g. old event
+      // registrations.
+      onNextRender: []
+    };
   },
-  
+
   resize: function(el, width, height, instance) {
     if (instance.autosize) {
       Plotly.relayout(el.id, {width: width, height: height});
     }
-  },  
-  
+  },
+
   renderValue: function(el, x, instance) {
-    
+
+    // Release previously registered crosstalk event listeners
+    while (instance.onNextRender.length > 0) {
+      instance.onNextRender.pop()();
+    }
+      
```

Comment:
No, I just made it up just now. It's totally self contained in this plotly.js file. If you can't tell what's going on, let me know and I'll write up more comments tomorrow.